### PR TITLE
Filling the help infrastructure

### PR DESCRIPTION
**Pull request by DelazJ**
This PR adds help button to some of the dialogs that missed it (ref https://github.com/qgis/qgis3_UIX_discussion/issues/33) and connects it to the online user manual. General implementation follows this structure (_hope that's correct_ - local build is fine):
- `.h` file: add `#include "qgshelp.h"` and the private slot `void showHelp();`
- `.cpp` file: connect the help button in the dialog initialisation and add the the showHelp function definition.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/app/qgsdecorationlayoutextentdialog.cpp [hunk 1]
```diff
@@ -42,6 +42,8 @@ QgsDecorationLayoutExtentDialog::QgsDecorationLayoutExtentDialog( QgsDecorationL
 
   updateGuiElements();
   connect( buttonBox->button( QDialogButtonBox::Apply ), &QAbstractButton::clicked, this, &QgsDecorationLayoutExtentDialog::apply );
+  connect( buttonBox, &QDialogButtonBox::helpRequested, this, &QgsDecorationLayoutExtentDialog::showHelp );
+// Pas écrit pareil ci-dessus
```
Comment (m-kuhn) [inline src/app/qgsdecorationlayoutextentdialog.cpp]: Pas écrit en mêne langue que le rèste du code

Comment (DelazJ) [inline src/app/qgsdecorationlayoutextentdialog.cpp]: Indeed 😄 
Btw, maybe a beginner question, what's the difference between the two ways the connect commands are written? Is any better? i tried to apply the second model to the "Apply" connection without success.

Comment (m-kuhn) [inline src/app/qgsdecorationlayoutextentdialog.cpp]: The first one directly connects to the button, the second one to a helper signal on the box. The second one only works with what's listed here under signals: http://doc.qt.io/qt-5/qdialogbuttonbox.html . I'm not aware of anythings that makes one of them preferable.

Comment (DelazJ) [inline src/app/qgsdecorationlayoutextentdialog.cpp]: Ah right... The ApplyRole isn't among and I now understand why my signal commands fail. Thanks.

File: src/app/qgsmapsavedialog.cpp [hunk 2]
```diff
@@ -459,3 +460,9 @@ void QgsMapSaveDialog::accepted()
     }
   }
 }
+
+void QgsMapSaveDialog::showHelp()
+{
+  // description needs to be written and i'm not sure of the place yet
```
Comment (m-kuhn) [inline src/app/qgsmapsavedialog.cpp]: Is this maybe better left as an issue on the tracker than in here?
I'm afraid that this comment will still be here in 10 years and no one knows why.

I think comments in the code should mainly help a reader understand the code or be very precise todos.

This also applies to all the `//to be precised` comments

Comment (DelazJ) [inline src/app/qgsmapsavedialog.cpp]: Removed. Will mention these information in the calc file that lists user manual links and buttons, for tracking.

Comment (m-kuhn) [inline src/app/qgsmapsavedialog.cpp]: Perfect, this sounds much more appropriate

**Pull request merged**